src/test/suite/completion.util.ts

```diff
@@ -27,14 +28,19 @@ export const getDocUri = (p: string) => {
 };
 
 export async function checkExpectedCompletion(docUri: vscode.Uri, position: vscode.Position, expectedCompletion: vscode.CompletionItem) {
-    let hasExpectedCompletion = false;
-    await waitUntil(() => {
-        // Executing the command `vscode.executeCompletionItemProvider` to simulate triggering completion
-        (vscode.commands.executeCommand('vscode.executeCompletionItemProvider', docUri, position)).then(value => {
-            let actualCompletionList = value as vscode.CompletionList;
-            const actualCompletionLabelList = actualCompletionList.items.map(c => { return c.label; });
-            hasExpectedCompletion = actualCompletionLabelList.includes(expectedCompletion.label);
-        });
-        return hasExpectedCompletion;
-    }, 10000, 500);
+	let hasExpectedCompletion = false;
+	let actualCompletionLabelList;
+	try {
+		await waitUntil(() => {
+			// Executing the command `vscode.executeCompletionItemProvider` to simulate triggering completion
+			(vscode.commands.executeCommand('vscode.executeCompletionItemProvider', docUri, position)).then(value => {
+				let actualCompletionList = value as vscode.CompletionList;
+				actualCompletionLabelList = actualCompletionList.items.map(c => { return c.label; });
+				hasExpectedCompletion = actualCompletionLabelList.includes(expectedCompletion.label);
+			});
+			return hasExpectedCompletion;
```

Comment:
I am not sure this is correct. It should be chained to the .then (is there a .finally?). Also you forgot a .catch() in case of an exception.  I would really suggest to stick with either async/await or promise chaining.

Reply:
see https://issues.redhat.com/browse/FUSETOOLS2-430